packages/core/src/connectors/google/index.ts

```diff
@@ -0,0 +1,139 @@
+import { accessTokenEndpoint, authorizationEndpoint, scope, userInfoEndpoint } from './constant';
+
+export const metadata: ConnectorMetadata = {
+  id: 'google',
+  type: ConnectorType.Social,
+  name: {
+    en: 'Sign In with Google',
+    zh_CN: 'Google登录',
+  },
+  logo: './logo.png',
+  description: {
+    en: 'Sign In with Google',
+    zh_CN: 'Google登录',
+  },
+};
+
+const googleConfigGuard = z.object({
+  clientId: z.string(),
+  clientSecret: z.string(),
+});
+
+type GoogleConfig = z.infer<typeof googleConfigGuard>;
+
+export const validateConfig: ValidateConfig = async (config: unknown) => {
+  if (!config) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, 'Missing config');
+  }
+
+  const result = googleConfigGuard.safeParse(config);
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error.message);
+  }
+};
+
+export const getAuthorizationUri: GetAuthorizationUri = async (redirectUri, state) => {
+  const config = await getConnectorConfig<GoogleConfig>(metadata.id);
+
+  return `${authorizationEndpoint}?${stringify({
+    client_id: config.clientId,
+    redirect_uri: redirectUri,
+    response_type: 'code',
+    state,
+    scope,
+  })}`;
+};
+
+export const getAccessToken: GetAccessToken = async (code, redirectUri) => {
+  type AccessTokenResponse = {
+    access_token: string;
+    scope: string;
+    token_type: string;
+  };
+
+  const { clientId, clientSecret } = await getConnectorConfig<GoogleConfig>(metadata.id);
+
+  // Note：Need to decodeURIComponent on code
+  // https://stackoverflow.com/questions/51058256/google-api-node-js-invalid-grant-malformed-auth-code
+  const { access_token: accessToken } = await got
+    .post(accessTokenEndpoint, {
+      form: {
+        code: decodeURIComponent(code),
+        client_id: clientId,
+        client_secret: clientSecret,
+        redirect_uri: redirectUri,
+        grant_type: 'authorization_code',
+      },
+      timeout: 5000,
+      followRedirect: true,
+    })
+    .json<AccessTokenResponse>();
+
+  if (!accessToken) {
+    throw new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid);
+  }
+
+  return accessToken;
+};
+
+export const getUserInfo: GetUserInfo = async (accessToken: string) => {
+  type UserInfoResponse = {
+    sub: string;
+    name: string;
+    given_name: string;
+    family_name: string;
+    picture: string;
+    email: string;
+    email_verified: boolean;
+    locale: string;
+  };
+
+  try {
+    const {
+      sub: id,
+      picture: avatar,
+      email,
+      name,
+    } = await got
+      .post(userInfoEndpoint, {
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        timeout: 5000,
+      })
+      .json<UserInfoResponse>();
+
+    return {
+      id,
+      avatar,
+      email,
+      name,
+    };
+  } catch (error: unknown) {
+    if (error instanceof GotRequestError && error.response?.statusCode === 401) {
```

Comment:
remind: this should be tested locally

Reply:
tested
<img width="471" alt="image" src="https://user-images.githubusercontent.com/10806653/156311002-06e2cba4-ca26-4b10-a035-920896e874ba.png">